jekyll-assets/scripts/theme.js

```diff
@@ -0,0 +1,32 @@
+// toggles the site theme between dark and light
+function toggleTheme() {
+  // fetch the theme from local storage (if it exists)
+  var theme = localStorage.getItem('theme');
+  // if the theme has never been set, or is light, set the theme to the dark symbol in local storage to change it
+  if (localStorage.getItem('theme') !== null && theme == '🌝' ) {
+    localStorage.setItem('theme', '🌚');
+  } else {
+    // otherwise, the theme is currently set to dark, so set the theme to the light symbol in local storage to change it
+    localStorage.setItem('theme', '🌝');
+  }
+  // finally, toggle the light option off or on the body to change the display of the theme
+  document.body.classList.toggle('light');
```

Comment:
Did you try setting `color-scheme: light` (and `color-scheme: dark`) and then relying solely on the `prefers-color-scheme` media query? If that works, I wonder if it’d make it simpler to implement dark mode support in the header and footer without relying on specific class names.

Reply:
Unfortunately a lot of browsers don't support the prefers-color-scheme query for anti-fingerprinting purposes. As a result, I'm much more comfortable with an explicit toggle persisted via a cookie or local storage. Using that query also has the unfortunate downside of assuming light mode as the default, which often results in that 'light flash' for dark mode users that I refer to in the comments!